### PR TITLE
fix: create finalization on REVERT opcode

### PR DESCRIPTION
**Pull request by enitrat**
Time spent on this PR:

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves #<Issue number>

## What is the new behavior?

- Fixes the finalization of a parent context in a CREATE execution. We need to handle the case where the EVM has errored (exceptional or classic revert) before anything else.
-
-

### CODE REVIEW

File: src/kakarot/instructions/system_operations.cairo [hunk 1]
```diff
@@ -1074,6 +1063,30 @@ namespace CreateHelper {
     }(evm: model.EVM*) -> model.EVM* {
         alloc_locals;
 
+        // Reverted during execution - either REVERT or exceptional
+        if (evm.reverted != FALSE) {
+            let is_exceptional_revert = is_not_zero(Errors.REVERT - evm.reverted);
```
Comment (Eikix) [inline src/kakarot/instructions/system_operations.cairo:1068]: why would Errors.REVERT be equal to 1

File: src/kakarot/instructions/system_operations.cairo [hunk 2]
```diff
@@ -1086,26 +1099,17 @@ namespace CreateHelper {
...
-            // The only case where a create opcode has returndata is if it reverted with the REVERT opcode.
-            tempvar is_exceptional_revert = is_not_zero(Errors.REVERT - evm.reverted) + (
-                1 - checks_success
-            );
-            let return_data_len = (1 - is_exceptional_revert) * evm.return_data_len;
-
             tempvar evm = new model.EVM(
                 message=evm.message.parent.evm.message,
-                return_data_len=return_data_len,
+                return_data_len=0,
```
Comment (Eikix) [inline src/kakarot/instructions/system_operations.cairo:1112]: what is happening in this case?
success is false, and revert is also false?

**Review by Eikix — COMMENTED** [review]
not sure how to validate this PR, except that more tests pass and none others are broken

File: src/kakarot/instructions/system_operations.cairo (hunk 1 above)
Comment (enitrat) [reply src/kakarot/instructions/system_operations.cairo:1068]: REVERT is 1, EXCEPTIONAL_HALT is 2

File: src/kakarot/instructions/system_operations.cairo (hunk 2 above)
Comment (enitrat) [reply src/kakarot/instructions/system_operations.cairo:1112]: the execution of the create was successful, but the checks before writing the code in the destination accounts were not.
